packages/block-editor/src/components/block-list/block.js

```diff
@@ -644,18 +644,34 @@ const applyWithDispatch = withDispatch( ( dispatch, ownProps, { select } ) => {
 		setAttributes( newAttributes ) {
 			const { name, clientId } = ownProps;
 			const type = getBlockType( name );
-			updateBlockAttributes( clientId, newAttributes );
-			const metaAttributes = reduce(
-				newAttributes,
-				( result, value, key ) => {
-					if ( get( type, [ 'attributes', key, 'source' ] ) === 'meta' ) {
-						result[ type.attributes[ key ].meta ] = value;
-					}
-
-					return result;
-				},
-				{}
-			);
+
```

Comment:
The behavior of all this setAttributes function seems like something that should be absorbed in the `updateBlockAttributes` action instead.

As you said, we should ship the bug fix and consider these consolidations separately.

Reply:
> The behavior of all this setAttributes function seems like something that should be absorbed in the `updateBlockAttributes` action instead.
> 
> As you said, we should ship the bug fix and consider these consolidations separately.

Yes, I think so too. One point of clarification / affirmation is that: I don't think the changes here dig us any further into a hole, and in-fact serve more as an implementation which ought to have existed in the first place.